### ISSUE TITLE
git-annex cannot build with the latest haskellNG + 7.10

### ISSUE THREAD

**Issue by jwiegley** [NixOS/nixpkgs#7527]
@peti With the latest master (465a8c6), and on Darwin (though I don't see any Darwin-specific issues), `git-annex` is unable to build due to failures in the following dependencies:

```
haskell-gnutls-0.1.5.drv
haskell-gsasl-0.3.5.drv
haskell-network-protocol-xmpp-0.4.6.drv
haskell-edit-distance-0.2.1.2.drv
haskell-feed-0.3.9.2.drv
haskell-DAV-1.0.3.drv
haskell-bencode-0.5.0.1.drv
haskell-torrent-10000.0.0.drv
```

My build log is here: https://gist.github.com/jwiegley/1062e9e550fceb5a1d19

In general, `git-annex` is an excellent litmus test, because it pulls in so many other packages.

**Comment by peti** [comment]
Yes, similar problems exist on Linux. IMHO the best way to remedy the situation is to report those issues upstream and/or to submit patches to the respective package authors to help them fix the problems with GHC 7.10.1. I don't believe that we should try to solve this is in isolation in Nixpkgs since none of those problems are specific to Nix.

**Comment by jwiegley** [comment]
@peti Sounds good.  Just FYI, here are the other Haskell packages I wasn't able to build with the latest, for various reasons:

```
haskell-CC-delcont-0.2
haskell-doctest-prop-0.2
haskell-recursion-schemes-4.1
haskell-liquidhaskell-0.3.1.0
haskell-compdata-0.9
haskell-machines-0.4.1
haskell-ghc-mod-5.2.1.2
haskell-arithmoi-0.4.1.2
splot-0.3.12
haskell-timeparsers-0.3.2
timeplot-1.0.28
haskell-time-recurrence-0.9.2
haskell-zip-archive-0.2.3.7
haskell-idris-0.9.17.1
haskell-singletons-1.1.1
haskell-thyme-0.3.5.5
```

I'll start narrowing the issues and reporting them upstream.
